yare-examples/src/test/java/com/sabre/oss/yare/example/StoppedActionTest.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.yare.example;
+
+import com.sabre.oss.yare.core.EngineController;
+import com.sabre.oss.yare.core.RuleSession;
+import com.sabre.oss.yare.core.RulesEngine;
+import com.sabre.oss.yare.core.RulesEngineBuilder;
+import com.sabre.oss.yare.core.model.Rule;
+import com.sabre.oss.yare.dsl.RuleDsl;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static com.sabre.oss.yare.dsl.RuleDsl.*;
+import static com.sabre.oss.yare.invoker.java.MethodCallMetadata.method;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class StoppedActionTest {
+
+    @Test
+    void shouldEvaluatingUpToTwoMatchingFacts() {
+        // given
+        List<Rule> rules = Collections.singletonList(RuleDsl.ruleBuilder()
+                .name("Should match airline when airline codes contain given")
+                .fact("airline", Airline.class)
+                .predicate(
+                        contains(
+                                castToCollection(value("${airline.airlineCodes}"), String.class),
+                                values(String.class, "AAU")
+                        )
+                )
+                .action("collect",
+                        param("context", value("${ctx}")),
+                        param("fact", value("${airline}")),
+                        param("engineController", value("${engineController}")))
+                .build());
+        List<Airline> facts = Arrays.asList(
+                new Airline().withAirlineCodes(
+                        Arrays.asList("AAU", "AAW", "AAV", "AFU")
+                ),
+                new Airline().withAirlineCodes(
+                        Collections.singletonList("AAU")
+                ),
+                new Airline().withAirlineCodes(
+                        Arrays.asList("PIU", "BRO")
+                ),
+                new Airline().withAirlineCodes(
+                        Arrays.asList("AAU", "PIU", "BRO")
+                ));
+
+        RulesEngine engine = new RulesEngineBuilder()
+                .withRulesRepository(i -> rules)
+                .withActionMapping("collect", method(new StoppedActionTest.Actions(2), (action) -> action.collectUpToTwo(null, null, null)))
+                .build();
+        RuleSession session = engine.createSession("airlines");
+
+        // when
+        List<Airline> matching = session.execute(new ArrayList<>(), facts);
+
+        // then
+        assertThat(matching).hasSize(2);
```

Comment:
Since you use below `containsExactly(...)` this assertion is redundant.

Reply:
Done.